core/src/main/java/com/graphhopper/routing/profiles/SimpleIntEncodedValue.java

```diff
@@ -107,7 +107,10 @@ public final void setInt(boolean reverse, IntsRef ref, int value) {
     }
 
     final void uncheckedSet(boolean reverse, IntsRef ref, int value) {
-        if (storeBothDirections && reverse) {
+        if (reverse && !storeBothDirections)
+            throw new IllegalArgumentException(getName() + ": value for reverse direction would overwrite forward direction. Enable 'storeBothDirections' or always use reverse==true when storing the value");
```

Comment:
Nice! Will this fix #1588 ?

Reply:
I think so. As it avoids the overwriting of the reverse property (setReverse throws an exception now). 

Do you think we should do the same for getReverse without bi-direction support? Currently I don't do this for convenience reasons (no need to do `if(encodedValue.isStoreBothDirections)` everytime you access a property and you want bi-directional support e.g. in a Weighting).

Reply:
> Do you think we should do the same for getReverse

I would say no. If you call `getReverse` even though its not bidirectional that's just fine you get the same value as if you had called `get`. I do not think this would be something unexpected. You do not know the value beforehand and calling `get` or `getReverse` gives you some value you need to trust. Maybe both directions are the same or the property is not bi-directional - this should not be of your concern when just calling the getter.

In comparison, calling `setReverse`, really gives you the impression that you can set a special reverse value, but if its not bidirectional it will be lost or even overwrite the other direction, which is a nasty surprise, so here I think it is good that it throws now.